File: Algorithm/BOJ/Silver/1244스위치켜고끄기.py
import sys

sys.stdin = open("1244.txt")

N = int(input())

swich = list(map(int, input().split()))
for i in range(N):
    if swich[i]:
        swich[i] = True
    else:
        swich[i] = False
S = int(input())

for i in range(S):
    mf, num = map(int, input().split())

    if mf == 1 :
        for j in range(N):
            if (j+1) % num == 0:
                swich[j] = not swich[j]

    else :
        swich[num-1] = not swich[num-1]
        for j in range(1, min(N-num+1, num)):
            if swich[num-1 + j] != swich[num -1- j]:
                break
            swich[num-1 + j] =  not swich[num - 1+ j]
            swich[num - 1 - j] = not swich[num - 1 - j]
cnt = 0
for i in range(N):
    cnt += 1
    print(int(swich[i]) , end = ' ')
    if not cnt % 20 :
        print()

File: Algorithm/C/algo/algo/1244스위치켜고끄기.cpp
//#include <iostream>
//#include <algorithm>
//
//using namespace std;
//int amain(void) {
//	int N, num, mf, S;
//	int bulb[100];
//
//	freopen("1244스위치켜기.txt", "r", stdin);
//
//	std::cin >> N;
//	for (int i = 0; i < N; i++) {
//		cin >> bulb[i];
//	}
//	
//	cin >> S;
//
//	for (int i = 0; i < S; i++) {
//		cin >> mf >> num;
//		if (mf == 1) {
//			for (int j = 0; j < N; j ++) {
//				if ((j + 1) % num == 0) {
//					bulb[j] = !bulb[j];
//				}
//			}
//		}
//		else {
//			bulb[num - 1] = !bulb[num - 1];
//			for (int j = 1; j < min(N - num + 1, num); j++) {
//				if (bulb[num - 1 + j] != bulb[num - 1 - j]) {
//					break;
//				}
//				bulb[num - 1 + j] = !bulb[num - 1 + j];
//				bulb[num - 1 - j] = !bulb[num - 1 - j];
//			}
//		}
//	}
//	int cnt = 0;
//	for (int i = 0; i < N; i++) {
//		cnt++;
//		cout << bulb[i] << " ";
//		if (!(cnt % 20)) {
//			cout << "\n";
//		}
//	}
//
//
//	return 0;
//}